### PR TITLE
chore(e2e/app): DRAFT deploy contracts to test SILK against

### PR DESCRIPTION
**Pull request by Zodomo**
This is a draft PR seeking commentary on the Go I wrote. This is my first attempt at writing anything in Go, so I explicitly welcome criticism, feedback, and suggestions. Please let me know what you like, what you don't, and feel free to ask about why I did things certain ways if it is unclear.

issue: none

### CODE REVIEW

File: e2e/app/mocksymbiotic.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+			// Store address in deployment struct
+			switch i {
+			case 0:
+				deployment.WstETH = addr
+			case 1:
+				deployment.RETH = addr
+			case 2:
+				deployment.METH = addr
+			}
```
[the remark this answers is not in the document]
Comment (corverroos) [reply e2e/app/mocksymbiotic.go:87]: indeed, or something like:
```
token := struct {
   setter func(common.Address)
   ..
}{
  {
    setter: func(addr common.Addrss) { deployment.WstETH = addr }
  }
  ....
}

token.SetAddress(addr)
```

File: e2e/app/mocksymbiotic.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+			log.Info(ctx, "Token deployed",
```
Comment (corverroos) [inline e2e/app/mocksymbiotic.go:89]: lets make this debug, info should only be main workflow steps, this is a sub-step

File: e2e/app/mocksymbiotic.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+		}
+
+		deployments[chain.ChainID] = deployment
+	}
+
+	return deployments, nil
+}
+
+func deploySymbioticVaults(ctx context.Context, def Definition, deployedTokens map[uint64]TokenDeployment) (map[uint64]SymbioticVaultDeployment, error) {
+	vaults := []string{
```
Comment (corverroos) [inline e2e/app/mocksymbiotic.go:105]: suggest combining this and tokens above into a single package-level struct.

**Comment by Zodomo** [comment]
closed as this was more or less me just experimenting with the go codebase

**Pull request closed without merging**